### DISCUSSION TITLE
Projects (beta): Group by on Kanban board to add swimlanes

### DISCUSSION THREAD

**Discussion by erichexter** [community/community#5654] [Projects and Issues] [answered]
I expected the group by function to add swimlanes to the kanban board view.  There are many use cases for adding swimlanes to a kanban board, the most common is having an expedited swimlane for critical bugs or other urgent work. I could see other uses at the project level to create swim lanes by repositories or by custom fields to allow work to be organized by the team that is executing said work. All the items in the swimlane would be used by that team. 

There are many use cases:  Here are some examples of kanban swimlanes.  To keep it simple I am linking to a microsoft version of swimlane docs, but it is a pretty standard feature of a lot of kanban boards.   https://docs.microsoft.com/en-us/azure/devops/boards/boards/expedite-work?view=azure-devops

**Comment by e0d** [comment]
I would also like swimlanes on boards.  Group by could provide that.  I wonder if it shouldn't carry over across views.

**Comment by peppy** [comment]
I'd like to use this kind of grouping to create one column per release cycle, to share a "roadmap" view with users.

**Comment by umisora** [comment]
We use the same feature in JIRA.
https://ja.confluence.atlassian.com/jirasoftwarecloud/configuring-swimlanes-764478010.html


Even when we are working on the smallest task in an AgileProject, we can see which Story we are working on. Are we in a hurry to start the next Story? We need Kanban + Swimlane to know if the team is working on the next Story in a hurry.

The Swimlane puts the SubTask in a box. This is much easier to recognize than a Label.

When the box is empty, the team can start working on the next Story. This is a very simple rule.

We are looking forward to your implementation.

![image](https://user-images.githubusercontent.com/16054205/146846658-f058cd6d-59f7-49bd-a181-98204e0d59be.png)

**Comment by itcarroll** [comment]
A "Grid" layout for views would be the logical offspring of "Table" (i.e. rows) and "Board" (i.e. columns) layouts.

**Comment by michaelw90** [comment]
Swimlane support would definitely be helpful and make it so that we can migrate our full workflow over to GitHub Projects, without much variation in terms of our current workflow, but makes it a friendlier Dev experience for our dev teams.

**Comment by itcarroll** [comment]
I just got on the waitlist for the [Tasklists](https://github.com/features/issues/signup) beta. Anyone here testing already? I imagine with the "tracked-by" filter you would have something closer to utility of swimlanes, albeit restricted to tracking relationships so not very general.

**Comment by erichexter** [comment]
thanks for mentioning that I haven't seen the Tasklists yet

**Comment by rileybroughten** [comment, marked as answer]
👋 This is coming! Check out the [roadmap issue](https://github.com/github/roadmap/issues/787) for more details, and note that this will land early Q3 (July-Sept), rather than Q2 (April-June) as noted ✨

**Reply by erichexter** [reply]
Thats awesome, thankyou for following up on this thread!

**Reply by hamzahc1** [reply]
Is there any update on when this might ship?

**Reply by rileybroughten** [reply]
This shipped last week! Check out the [changelog](https://github.blog/changelog/2023-07-27-github-issues-projects-july-27th-update/#%F0%9F%8F%8A-board-swimlanes) for more details ✨

**Reply by erichexter** [reply]
great works as expected, I added a new field and it allowed grouping by that new field.!!

**Comment by erichexter** [comment]
This is deployed and in the product.. closing discussion.
